src/apple/xcode-manager.ts

```diff
@@ -302,103 +331,971 @@ export class XcodeProject {
+    // - The target defines the list of `fileSystemSynchronizedGroups` which are `PBXFileSystemSynchronizedRootGroup` to be included in the build phase
+    // - The `PBXFileSystemSynchronizedRootGroup` has a list of `exceptions` which are `PBXFileSystemSynchronizedBuildFileExceptionSet`
+    // - Each `PBXFileSystemSynchronizedBuildFileExceptionSet` represents a folder to be excluded from the build.
+    // - The `PBXFileSystemSynchronizedBuildFileExceptionSet` has a list of `membershipExceptions` which are files to be excluded from being excluded, therefore included in the build.
+    // - The Xcode project has a build phase `PBXSourcesBuildPhase` which has a list of `files` which are `PBXBuildFile`
+    // - A file which is not part of a `PBXFileSystemSynchronizedRootGroup` must be added to the `files` list of the `PBXSourcesBuildPhase` build phase
+    // - Nested subfolders in `fileSystemSynchronizedGroups` are not declared but recursively included
+    //
+    // Based on the findings above the files included in the build phase are:
+    // - All files in the `files` of the `PBXSourcesBuildPhase` build phase `Sources` of the target
+    // - All files in directories of the `fileSystemSynchronizedGroups` of the target
+    // - Excluding all files in the `exceptions` of the `PBXFileSystemSynchronizedRootGroup` of the target
+    // - Including all files in the `membershipExceptions` of the `PBXFileSystemSynchronizedBuildFileExceptionSet` of the target
+    const nativeTarget = this.findNativeTargetByName(targetName);
+    if (!nativeTarget) {
+      debug('Target not found: ' + targetName);
+      return undefined;
+    }
 
-    const targets = this.objects.PBXNativeTarget || {};
-    const nativeTarget = Object.keys(targets).filter((key) => {
-      const value = targets[key];
-      return (
-        !key.endsWith('_comment') &&
-        typeof value !== 'string' &&
-        value.name === target
+    const filesInBuildPhase = this.findFilesInSourceBuildPhase(nativeTarget);
+    debug(
+      `Found ${filesInBuildPhase.length} files in build phase for target: ${targetName}`,
+    );
+
+    const filesInSynchronizedRootGroups =
+      this.findFilesInSynchronizedRootGroups(nativeTarget);
+    debug(
+      `Found ${filesInSynchronizedRootGroups.length} files in synchronized root groups for target: ${targetName}`,
+    );
+    return [...filesInBuildPhase, ...filesInSynchronizedRootGroups];
+  }
+
+  // ================================ TARGET HELPERS ================================
+
+  /**
+   * Finds a native target by name.
+   *
+   * @param targetName - The name of the target to find
+   * @returns The native target, or undefined if the target is not found
+   */
+  private findNativeTargetByName(
+    targetName: string,
+  ): XcodeProjectObjectWithId<PBXNativeTarget> | undefined {
+    debug('Finding native target by name: ' + targetName);
+
+    const nativeTargets = Object.entries(this.objects.PBXNativeTarget ?? {});
```

Comment:
`l`: We could add an early return if the `nativeTargets` are empty and include a log a debug message.